calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,397 @@
+            m_check_output.assert_called_once_with([
+                'docker', 'inspect', '-format', '{{ .NetworkSettings.IPAddress }}',
+                self.plugin.docker_id])
+            self.assertEqual(return_val, '1.2.3.4')
+
+    def test_delete_docker_interface(self):
+        with patch.object(calico_kubernetes, 'check_output',
+                    autospec=True) as m_check_output:
+            # Set up mock objects
+            m_check_output.return_value = 'pid'
+
+            # Call method under test
+            self.plugin._delete_docker_interface()
+
+            # Assert call list
+            call_1 = call([
+                'docker', 'inspect', '-format', '{{ .State.Pid }}',
+                self.plugin.docker_id])
+            call_2 = call(['mkdir', '-p', '/var/run/netns'])
+            call_3 = call(['ln', '-s', '/proc/' + 'pid' + '/ns/net',
+                            '/var/run/netns/pid'])
+            call_4 = call(['ip', 'netns', 'exec', 'pid', 'ip', 'link', 'del', 'eth0'])
+            call_5 = call(['rm', '/var/run/netns/pid'])
+            calls = [call_1,call_2,call_3,call_4,call_5]
+
+            m_check_output.assert_has_calls(calls)
+
+    def test_configure_profile(self):
+        with patch.object(self.plugin, 'calicoctl',
+                    autospec=True) as m_calicoctl, \
+                patch.object(self.plugin, '_get_pod_config',
+                    autospec=True) as m_get_pod_config, \
+                patch.object(self.plugin, '_apply_rules',
+                    autospec=True) as m_apply_rules, \
+                patch.object(self.plugin, '_apply_tags',
+                    autospec=True) as m_apply_tags:
+            # Set up mock objects
+            m_endpoint = Mock()
+            m_endpoint.endpoint_id = 'ep_id'
+            m_get_pod_config.return_value = 'pod'
+
+            # Set up class members
+            self.plugin.pod_name = 'podname'
+
+            # Call method under test
+            self.plugin._configure_profile(m_endpoint)
+
+            # Assert
+            m_calicoctl_call_1 = call('profile', 'add', 'podname')
+            m_calicoctl_call_2 = call('endpoint', 'ep_id', 'profile',
+                                        'set', self.plugin.pod_name)
+            m_calicoctl_calls = [m_calicoctl_call_1,m_calicoctl_call_2]
+            m_calicoctl.assert_has_calls(m_calicoctl_calls)
+            m_get_pod_config.assert_called_once_with()
+            m_apply_rules.assert_called_once_with('podname')
+            m_apply_tags.assert_called_once_with('podname', 'pod')
+
+    def test_get_pod_ports(self):
+        # Initialize pod dictionary and expected outcome
+        pod = {'spec': {'containers': [{'ports': [1, 2, 3]},{'ports': [4, 5]}]}}
+        ports = [1, 2, 3, 4, 5]
+
+        # Call method under test
+        return_val = self.plugin._get_pod_ports(pod)
+
+        # Assert
+        self.assertEqual(return_val, ports)
+
+    def test_get_pod_ports_fail(self):
```

Comment:
For failure tests especially, its good to indicate what kind of failure they are trying to reproduce (in a docstring / comment).

If we come back and take a look at this guy in 5 months we might have no clue what it is trying to test.